In [1]:
import pandas as pd

df = pd.read_csv('../data/European_Bank.csv')
df.head()

,Year,CustomerId,Surname,CreditScore,Geography,Gender,Age,Tenure,Balance,NumOfProducts,HasCrCard,IsActiveMember,EstimatedSalary,Exited
0,2025,15634602,Hargrave,619,France,Female,42,2,0.00,1,1,1,101348.88,1
1,2025,15647311,Hill,608,Spain,Female,41,1,83807.86,1,0,1,112542.58,0
2,2025,15619304,Onio,502,France,Female,42,8,159660.80,3,1,0,113931.57,1
3,2025,15701354,Boni,699,France,Female,39,1,0.00,2,0,0,93826.63,0
4,2025,15737888,Mitchell,850,Spain,Female,43,2,125510.82,1,1,1,79084.10,0


In [2]:
print(df.shape)
df.dtypes

(10000, 14)


Year                 int64
CustomerId           int64
Surname                str
CreditScore          int64
Geography              str
Gender                 str
Age                  int64
Tenure               int64
Balance            float64
NumOfProducts        int64
HasCrCard            int64
IsActiveMember       int64
EstimatedSalary    float64
Exited               int64
dtype: object

In [3]:
df['Year'].value_counts()

Year
2025    10000
Name: count, dtype: int64

In [4]:
print(df.isnull().sum())
print("Full row duplicates:", df.duplicated().sum())
print("Duplicate CustomerIds:", df['CustomerId'].duplicated().sum())

Year               0
CustomerId         0
Surname            0
CreditScore        0
Geography          0
Gender             0
Age                0
Tenure             0
Balance            0
NumOfProducts      0
HasCrCard          0
IsActiveMember     0
EstimatedSalary    0
Exited             0
dtype: int64
Full row duplicates: 0
Duplicate CustomerIds: 0


In [5]:
for col in ['HasCrCard', 'IsActiveMember', 'Exited']:
    print(col, df[col].unique())

HasCrCard [1 0]
IsActiveMember [1 0]
Exited [1 0]


In [6]:
df[['CreditScore','Age','Tenure','Balance','EstimatedSalary']].describe()

,CreditScore,Age,Tenure,Balance,EstimatedSalary
count,10000.000000,10000.000000,10000.000000,10000.000000,10000.000000
mean,650.528800,38.921800,5.012800,76485.889288,100090.239881
std,96.653299,10.487806,2.892174,62397.405202,57510.492818
min,350.000000,18.000000,0.000000,0.000000,11.580000
25%,584.000000,32.000000,3.000000,0.000000,51002.110000
50%,652.000000,37.000000,5.000000,97198.540000,100193.915000
75%,718.000000,44.000000,7.000000,127644.240000,149388.247500
max,850.000000,92.000000,10.000000,250898.090000,199992.480000


In [7]:
zero_balance = (df['Balance'] == 0).sum()
print(f"Customers with zero balance: {zero_balance} ({zero_balance/len(df)*100:.1f}%)")

Customers with zero balance: 3617 (36.2%)


In [8]:
print(df['Geography'].unique())
print(df['Gender'].unique())

<ArrowStringArray>
['France', 'Spain', 'Germany']
Length: 3, dtype: str
<ArrowStringArray>
['Female', 'Male']
Length: 2, dtype: str


In [9]:
df['Exited'].value_counts(normalize=True)

Exited
0    0.7963
1    0.2037
Name: proportion, dtype: float64

## Data Validation Summary

- Dataset contains 10,000 customer records, 14 columns, no missing values, no duplicate rows or Customer Ids.
- 'Year' is constant across all rows. Column will be excluded from analysis.
- Binary fields are encoded as 0/1.
- Numeric fields (CreditScore, Age, Tenure, Balance, EstimatedSalary) fall within plausible real-world ranges
- 36.2% of customers hold a zero balance, no missing data.
- Overall churn rate is 20.37% — this is the baseline against which all churn rates will be compared.
- Dataset is clean and requires no imputation or deduplication before proceeding to analysis.